kernel/constant_typing.ml

```diff
@@ -281,20 +275,23 @@ let check_delayed (type a) (handle : a effect_handler) tyenv (body : a proof_out
        Opaqueproof.PrivatePolymorphic private_univs
   in
   (* Note: non-trivial trusted side-effects only in monomorphic case *)
-  let body,env,ectx = skip_trusted_seff valid_signatures body env in
   let hbody = HConstr.of_constr env body in
-  let j = Typeops.infer_hconstr env hbody in
-  let j = unzip ectx j in
-  let _ = Typeops.judge_of_cast env j DEFAULTcast tyj in
+  let body, env = skip_trusted_seff valid_signatures hbody env in
```

Comment:
The name usage between `body` and `hbody` feels confusing, maybe here should be `let body_skipped_seff, env = ...`
and maybe `hbody` can be just `body`

alternatively this one stays `body` and `hbody` is `body_with_seff`??

or whatever you can come up with

Reply:
Fixed. The scope of the term without side-effect is now restricted to the typing component. Note that there was probably a bug before, since we would check declared variables w.r.t. the term without side-effects.